elrond-codec/src/impl_for_types/impl_vec.rs

```diff
@@ -1,3 +1,5 @@
+#![allow(clippy::transmute_undefined_repr)]
```

Comment:
Only add this on the respective methods, not the entire crate, I think the warning can be useful.
I will investigate more if there are better solutions to the layout problem.